flink-python/pyflink/datastream/tests/test_data_stream.py

```diff
@@ -455,6 +455,56 @@ def filter(self, value):
         expected = ['+I[c, 1]', '+I[e, 2]']
         self.assert_equals_sorted(expected, results)
 
+    def test_keyed_sum_with_tuple_type(self):
+        ds = self.env.from_collection([('a', 1), ('a', 2), ('a', 3), ('b', 1), ('b', 2)],
+                                      type_info=Types.TUPLE([Types.STRING(), Types.INT()]))
+        keyed_stream = ds.key_by(lambda x: x[0], key_type=Types.STRING())
+
+        keyed_stream.sum(1)\
+            .add_sink(self.test_sink)
+        self.env.execute('key_by_sum_test_with_tuple_type')
+        results = self.test_sink.get_results(False)
+        if self.__class__ == StreamingModeDataStreamTests:
```

Comment:
What about splitting the test cases into StreamingModeDataStreamTests and BatchModeDataStreamTests separately and then we could avoid this check.